apis/v1alpha1/nginxproxy_types.go

```diff
@@ -0,0 +1,113 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +kubebuilder:object:root=true
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+
+// NginxProxy represents the dynamic configuration for an NGINX Gateway Fabric data plane.
+type NginxProxy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// NginxProxySpec defines the desired state of the NginxProxy.
+	Spec NginxProxySpec `json:"spec"`
+
+	// NginxProxyStatus defines the state of the NginxProxy.
+	Status NginxProxyStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// NginxProxyList contains a list of NginxProxies.
+type NginxProxyList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []NginxProxy `json:"items"`
+}
+
+// NginxProxySpec defines the desired state of the NginxProxy.
+type NginxProxySpec struct {
+	HTTP *HTTP `json:"http,omitempty"`
+}
+
+// Http defines the Http configuration.
+type HTTP struct {
+	Telemetry *Telemetry `json:"telemetry,omitempty"`
+}
+
+// Telemetry defines the telemetry configuration.
+type Telemetry struct {
+	Tracing *Tracing `json:"tracing,omitempty"`
+}
+
+// TODO(ciarams87): need to figure out why the pattern validation for endpoint isn't adding to schema.
+
+// Tracing defines the tracing configuration.
+type Tracing struct {
+	// Endpoint specifies the address of OTLP/gRPC endpoint that will accept telemetry data.
+	//
+	// +required
+	// ++kubebuilder:validation:Required
+	// ++kubebuilder:validation:Pattern=`^(?:(?:[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?\.)+[a-zA-Z]{2,}|(?:\d{1,3}\.){3}\d{1,3}):\d{1,5}$`
+	Endpoint string `json:"endpoint"`
+	// Interval specifies the tracing interval. Default is 5s.
+	//
+	// +optional
+	// +kubebuilder:default="5s"
+	// +kubebuilder:validation:Pattern=`^(\d+y)??\s*(\d+M)??\s*(\d+w)??\s*(\d+d)??\s*(\d+h)??\s*(\d+m)??\s*(\d+s?)??\s*(\d+ms)??$`
+	Interval string `json:"interval,omitempty"`
+	// BatchSize specifies the maximum number of spans to be sent in one batch per worker. Default is 512.
+	//
+	// +optional
+	// +kubebuilder:default=512
+	BatchSize int `json:"batchSize,omitempty"`
```

Comment:
I believe k8s recommends integer types be int32, so they aren't ambiguous in their size.